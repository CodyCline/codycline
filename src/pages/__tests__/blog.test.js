import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from "gatsby"
import Blog from "../blog";

beforeEach(() => {
    StaticQuery.mockImplementationOnce(({ render }) =>
        render({
            site: {
                siteMetadata: {
                    title: `Default Starter`,
                },
            },
        })
    )
})
describe("Index", () => {
    it("renders correctly", () => {
        const data = {
            allMdx: {
                edges: [
                    {
                        node: {
                            fields: { slug: "/blog/diy-randonautica/" },
                            frontmatter: {
                                description: "Walkthrough on how to build a web app that gives you random locations to ... explore?",
                                tags: ["python", "mystery", "react"],
                                thumb: { publicURL: "/static/cd989575e12274e973f0f48f8ab96111/randonautica.webp" },
                                title: "Create your own Randonautica web app",
                            }
                        },
                        id: "3c8dc015-f590-5dd1-9dd7-82dd96215778"
                    }
                ]
            },
            site: {
                siteMetadata: {
                    title: "Cody Cline"
                }
            }
        }
        const tree = renderer.create(<Blog data={data} />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})