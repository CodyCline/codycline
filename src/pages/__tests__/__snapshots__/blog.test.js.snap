// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="nav__item home__icon"
      href="/"
    >
      <img
        alt="logo.png"
        src="test-file-stub"
        style={
          Object {
            "height": "50px",
            "width": "70px",
          }
        }
      />
    </a>
    <a
      className="nav__item"
      href="/blog"
    >
       Blog 
    </a>
    <a
      className="nav__item"
      href="/snippets"
    >
      Snippets
    </a>
    <a
      className="nav__item"
      href="/projects"
    >
      Projects
    </a>
    <a
      className="nav__item controls"
      href="#"
    >
      <button
        className="theme__toggle"
        onClick={[Function]}
      >
        <svg
          className="theme__toggle-icon"
          height={32}
          viewBox="0 0 1024 1024"
          width={32}
        >
          <title>
            lightbulb
          </title>
          <path
            d="M746.944 554.116c-21.683 21.347-38.349 45.102-50.054 71.274-11.398 26.079-17.126 54.412-17.126 85.068v102.419c0 11.488-2.17 22.342-6.541 32.506-4.32 10.048-10.253 18.861-17.805 26.381-7.546 7.52-16.385 13.427-26.46 17.734-10.187 4.346-21.067 6.502-32.575 6.502h-102.796c-11.509 0-22.389-2.157-32.576-6.502-10.074-4.307-18.913-10.214-26.461-17.734s-13.482-16.333-17.804-26.381c-4.371-10.163-6.538-21.018-6.538-32.506v-102.419c0-30.65-5.859-58.97-17.519-85.044-11.437-26.159-27.981-49.954-49.664-71.298-27.83-27.458-49.316-58.114-64.379-91.95-15.107-33.937-22.631-70.696-22.631-110.196 0-26.425 3.419-51.937 10.271-76.516 6.835-24.517 16.549-47.431 29.149-68.723 12.584-21.53 27.637-41.028 45.153-58.481 17.508-17.445 36.936-32.44 58.271-44.977 21.622-12.544 44.769-22.236 69.365-29.044 24.658-6.824 50.251-10.229 76.76-10.229s52.101 3.405 76.759 10.229c24.598 6.808 47.587 16.484 68.95 29.036 21.6 12.536 41.171 27.532 58.688 44.986 17.517 17.452 32.563 36.95 45.152 58.48 12.602 21.292 22.317 44.206 29.146 68.724 6.854 24.579 10.272 50.091 10.272 76.516 0 39.5-7.52 76.259-22.63 110.196-15.059 33.836-36.55 64.492-64.378 91.95zM615.8 672h-141.629v140.877c0 5.19 1.826 9.6 5.774 13.536 3.949 3.93 8.396 5.766 13.643 5.766h102.796c5.247 0 9.694-1.837 13.643-5.766 3.948-3.936 5.774-8.346 5.774-13.536v-140.877z"
            fill={null}
            opacity={1}
          />
        </svg>
      </button>
    </a>
  </nav>,
  <div
    style={
      Object {
        "height": "10vh",
        "width": "undefinedvh",
      }
    }
  />,
  <div
    className="content__container"
  >
    <div
      className="content__header"
    >
      <h2
        className="no__margin"
      >
        Blog Posts
      </h2>
      <p>
        Full feature articles
      </p>
      <hr
        className="divider"
      />
    </div>
    <article
      className="panel"
    >
      <a
        className="panel__image"
        href="/blog/diy-randonautica/"
      >
        <img
          alt="thumb.jpg"
          className="panel__image"
          role="link"
          src="/static/cd989575e12274e973f0f48f8ab96111/randonautica.webp"
        />
      </a>
      <h3
        className="panel__header"
      >
        <a
          href="/blog/diy-randonautica/"
        >
          Create your own Randonautica web app
        </a>
      </h3>
      <p
        className="panel__text"
      >
        <a
          href="/blog/diy-randonautica/"
        >
          Walkthrough on how to build a web app that gives you random locations to ... explore?
        </a>
      </p>
      <div
        className="panel__links"
      >
        <a
          className="meta"
          href="/meta/python"
        >
          <svg
            height={24}
            viewBox="0 0 1024 1024"
            width={24}
          >
            <title>
              python
            </title>
            <path
              d="M510.971 0.003c-261.316 0-244.998 113.322-244.998 113.322l0.291 117.401h249.368v35.249h-348.416c0 0-167.216-18.964-167.216 244.707s145.95 254.32 145.95 254.32h87.104v-122.353c0 0-4.695-145.95 143.619-145.95s247.328 0 247.328 0c0 0 138.958 2.246 138.958-134.297s0-225.771 0-225.771c0-0 21.098-136.628-251.989-136.628zM373.469 78.95c24.81-0 44.863 20.053 44.863 44.863s-20.053 44.863-44.863 44.863c-24.81 0-44.863-20.053-44.863-44.863s20.053-44.863 44.863-44.863z"
              fill="rgb(69, 132, 182)"
              opacity={1}
            />
            <path
              d="M518.392 1024.005c261.316 0 244.998-113.322 244.998-113.322l-0.291-117.401h-249.368v-35.249h348.416c0 0 167.216 18.964 167.216-244.707s-145.95-254.32-145.95-254.32h-87.104v122.353c0 0 4.695 145.95-143.619 145.95s-247.328 0-247.328 0c0 0-138.958-2.246-138.958 134.297s0 225.771 0 225.771c0 0-21.098 136.628 251.989 136.628zM655.894 945.058c-24.81 0-44.863-20.053-44.863-44.863s20.053-44.863 44.863-44.863c24.81 0 44.863 20.053 44.863 44.863s-20.053 44.863-44.863 44.863z"
              fill="rgb(255, 222, 87)"
              opacity={1}
            />
          </svg>
          <span
            className="meta__tag"
          >
            python
          </span>
        </a>
        <a
          className="meta"
          href="/meta/mystery"
        >
          <svg
            height={24}
            viewBox="0 0 1024 1024"
            width={24}
          >
            <title>
              mystery
            </title>
            <path
              d="M404.042 0c-159.638 0-263.036 65.406-344.214 182.052-14.726 21.16-10.186 50.172 10.356 65.748l86.276 65.418c20.746 15.73 50.264 12.052 66.506-8.296 50.098-62.762 87.26-98.898 165.514-98.898 61.528 0 137.632 39.598 137.632 99.262 0 45.104-37.234 68.268-97.986 102.328-70.846 39.72-164.598 89.152-164.598 212.81v19.576c0 26.51 21.49 48 48 48h144.942c26.51 0 48-21.49 48-48v-11.546c0-85.72 250.536-89.29 250.536-321.254 0.002-174.688-181.202-307.2-350.964-307.2zM384 746.918c-76.392 0-138.542 62.15-138.542 138.542 0 76.39 62.15 138.54 138.542 138.54s138.542-62.15 138.542-138.542-62.15-138.54-138.542-138.54z"
              fill="#CCCCCC"
              opacity={1}
            />
          </svg>
          <span
            className="meta__tag"
          >
            mystery
          </span>
        </a>
        <a
          className="meta"
          href="/meta/react"
        >
          <svg
            height={24}
            viewBox="0 0 1024 1024"
            width={24}
          >
            <title>
              react
            </title>
            <path
              d="M592 511.168c0 44.183-35.817 80-80 80s-80-35.817-80-80c0-44.183 35.817-80 80-80s80 35.817 80 80z"
              fill="rgb(96, 217, 251)"
              opacity={1}
            />
            <path
              d="M512 694.592c-6.541 0.169-14.241 0.266-21.965 0.266-93.518 0-183.735-14.133-268.632-40.379l6.436 1.713c-44.908-14.477-83.83-35.839-117.713-63.194l0.689 0.538c-26.177-18.965-43.806-48.345-46.785-81.936l-0.031-0.432c0-52.896 58.112-104.736 155.456-138.656 83.302-27.423 179.182-43.239 278.755-43.239 4.848 0 9.688 0.037 14.518 0.112l-0.729-0.009c3.812-0.057 8.311-0.090 12.819-0.090 98.524 0 193.421 15.528 282.378 44.271l-6.492-1.813c43.362 14.053 80.98 34.388 114.005 60.309l-0.789-0.597c25.754 18.111 43.144 46.621 46.050 79.297l0.030 0.415c0 54.976-64.96 110.688-169.6 145.312-78.648 24.269-169.060 38.245-262.737 38.245-5.507 0-11.003-0.048-16.487-0.145l0.824 0.011zM512 367.648c-3.555-0.051-7.751-0.080-11.953-0.080-95.704 0-187.896 15.032-274.345 42.865l6.33-1.761c-89.6 31.264-129.76 74.016-129.76 102.4 0 29.696 43.168 76.384 137.952 108.8 74.875 23.297 160.961 36.716 250.181 36.716 7.594 0 15.165-0.097 22.712-0.29l-1.117 0.023c4.698 0.090 10.239 0.141 15.791 0.141 89.34 0 175.582-13.265 256.864-37.933l-6.256 1.632c98.4-32.608 143.2-79.36 143.2-108.992-3.44-21.117-15.347-38.937-32.053-50.167l-0.267-0.169c-28.817-22.454-62.428-40.483-98.844-52.281l-2.308-0.647c-78.69-25.594-169.237-40.351-263.231-40.351-4.535 0-9.061 0.034-13.58 0.103l0.683-0.008z"
              fill="rgb(96, 217, 251)"
              opacity={1}
            />
            <path
              d="M330.24 910.176c-0.577 0.014-1.256 0.022-1.937 0.022-14.977 0-29.044-3.899-41.241-10.738l0.426 0.219c-45.824-26.432-61.696-102.656-42.464-203.936 23.157-103.929 60.719-195.822 111.108-279.085l-2.308 4.109c49.786-88.214 109.957-163.688 180.243-228.335l0.653-0.593c30.675-28.039 66.535-51.054 105.875-67.369l2.413-0.887c13.49-6.473 29.328-10.255 46.048-10.255s32.557 3.782 46.703 10.536l-0.655-0.281c47.648 27.456 63.424 111.552 41.184 219.488-23.911 98.556-60.394 185.343-108.418 264.339l2.306-4.083c-48.16 87.38-106.484 162.197-174.777 226.551l-0.455 0.425c-31.803 29.148-69.194 52.829-110.294 69.19l-2.442 0.858c-15.405 5.898-33.217 9.471-51.817 9.822l-0.151 0.002zM386.976 439.776c-45.828 75.721-81.813 163.576-103.166 257.004l-1.154 6.004c-17.696 93.28-0.704 149.376 24 163.648 25.6 14.816 87.744 0.768 163.2-65.152 65.608-62.034 121.282-133.651 164.994-212.798l2.27-4.482c43.797-71.622 78.785-154.588 100.528-242.769l1.264-6.063c20.928-101.6 2.848-163.808-22.816-178.592-7.906-3.12-17.063-4.93-26.643-4.93-12.073 0-23.474 2.873-33.559 7.973l0.426-0.195c-37.144 15.551-69.031 36.23-96.547 61.583l0.227-0.207c-67.761 62.473-125.241 134.727-170.523 214.792l-2.277 4.376z"
              fill="rgb(96, 217, 251)"
              opacity={1}
            />
            <path
              d="M693.664 910.592c-43.36 0-98.432-26.24-155.776-75.552-72.231-65.893-133.411-142.264-181.523-227.025l-2.381-4.559c-47.457-78.029-84.699-168.576-106.778-264.905l-1.19-6.167c-5.394-23.485-8.486-50.453-8.486-78.144 0-17.584 1.247-34.878 3.657-51.796l-0.227 1.94c2.926-33.013 20.238-61.466 45.548-79.356l0.34-0.228c47.584-27.552 128.32 0.768 210.72 73.92 67.213 63.595 124.547 136.671 170.092 217.291l2.388 4.597c47.439 76.285 84.989 164.759 107.847 258.975l1.273 6.209c5.422 23.463 8.529 50.406 8.529 78.075 0 19.27-1.507 38.187-4.41 56.641l0.265-2.045c-3.439 33.936-21.385 63.092-47.453 81.522l-0.355 0.238c-11.582 6.538-25.43 10.39-40.178 10.39-0.669 0-1.336-0.008-2.001-0.024l0.099 0.002zM387.104 584.192c48.461 85.41 106.903 158.5 175.096 221.141l0.648 0.587c71.968 61.92 129.056 75.232 153.6 60.8 25.6-14.88 44.48-75.616 25.024-173.888-23.273-95.892-59.276-180.331-106.816-256.856l2.144 3.704c-45.759-81.59-100.491-151.565-164.256-212.118l-0.384-0.362c-77.568-68.864-140.48-84.256-166.112-69.408-15.675 12.81-25.988 31.671-27.349 52.966l-0.011 0.218c-1.835 12.997-2.882 28.011-2.882 43.269 0 25.097 2.834 49.531 8.2 73.002l-0.421-2.19c22.382 98.023 58.092 184.593 105.753 263.104l-2.201-3.904z"
              fill="rgb(96, 217, 251)"
              opacity={1}
            />
          </svg>
          <span
            className="meta__tag"
          >
            react
          </span>
        </a>
      </div>
    </article>
  </div>,
  <div
    style={
      Object {
        "height": "60vh",
        "width": "undefinedvh",
      }
    }
  />,
  <footer
    className="footer"
  >
    <li
      className="footer__item"
    >
      2021
    </li>
    <li
      className="footer__item"
    />
  </footer>,
]
`;
